dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/DependencyInjectionTests.cs

```diff
@@ -56,64 +69,104 @@ public void CollectionNameCantBeNullOrEmpty()
+            // And the VectorStoreCollection abstraction.
+            Verify<VectorStoreCollection<TKey, TRecord>>(serviceProvider, lifetime, serviceKey);
+            // And the IVectorSearchable abstraction.
+            Verify<IVectorSearchable<TRecord>>(serviceProvider, lifetime, serviceKey);
+        }
     }
 
     [Theory]
     [MemberData(nameof(LifetimesAndServiceKeys))]
     public virtual void CanRegisterConcreteTypeVectorStoreAfterSomeAbstractionHasBeenRegistered(ServiceLifetime lifetime, object? serviceKey)
     {
-        IServiceCollection services = this.CreateServices(serviceKey);
+        foreach (var registrationDelegate in this.StoreDelegates)
+        {
+            IServiceCollection services = this.CreateServices(serviceKey);
 
-        // Users may be willing to register more than one IVectorStore implementation.
-        services.Add(new ServiceDescriptor(typeof(VectorStore), serviceKey, (sp, key) => new FakeVectorStore(), lifetime));
+            // Users may be willing to register more than one IVectorStore implementation.
+            services.Add(new ServiceDescriptor(typeof(VectorStore), serviceKey, (sp, key) => new FakeVectorStore(), lifetime));
 
-        this.RegisterVectorStore(services, lifetime, serviceKey);
+            registrationDelegate(services, serviceKey, lifetime);
 
-        using ServiceProvider serviceProvider = services.BuildServiceProvider();
-        // let's ensure that concrete types are registered
-        Verify<TVectorStore>(serviceProvider, lifetime, serviceKey);
+            using ServiceProvider serviceProvider = services.BuildServiceProvider();
+            // let's ensure that concrete types are registered
+            Verify<TVectorStore>(serviceProvider, lifetime, serviceKey);
+        }
     }
 
     [Theory]
     [MemberData(nameof(LifetimesAndServiceKeys))]
     public void CanRegisterConcreteTypeCollectionsAfterSomeAbstractionHasBeenRegistered(ServiceLifetime lifetime, object? serviceKey)
     {
-        IServiceCollection services = this.CreateServices(serviceKey);
+        foreach (var registrationDelegate in this.CollectionDelegates)
+        {
+            IServiceCollection services = this.CreateServices(serviceKey);
+
+            // Users may be willing to register more than one VectorStoreCollection implementation.
+            services.Add(new ServiceDescriptor(typeof(VectorStoreCollection<TKey, TRecord>), serviceKey, (sp, key) => new FakeVectorStoreRecordCollection(), lifetime));
+
+            registrationDelegate(services, serviceKey, "collectionName", lifetime);
+
+            using ServiceProvider serviceProvider = services.BuildServiceProvider();
+            // let's ensure that concrete types are registered
+            Verify<TCollection>(serviceProvider, lifetime, serviceKey);
+        }
+    }
+
+    [Theory]
+    [MemberData(nameof(LifetimesAndServiceKeys))]
+    public void EmbeddingGeneratorIsResolved(ServiceLifetime lifetime, object? serviceKey)
```

Comment:
Just noting that proper resolution of IEmbeddingGenerator from DI is tested end-to-end in EmbeddingGenerationTests (but it's good to have this unit test here too)